clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/config/CloudDriverConfig.groovy

```diff
@@ -296,19 +298,23 @@ class CloudDriverConfig {
   }
 
   @Bean
-  @ConditionalOnExpression('${redis.enabled:true}')
-  CoreProvider coreProvider(RedisCacheOptions redisCacheOptions,
-                            RedisClientDelegate redisClientDelegate,
+  CoreProvider coreProvider(Optional<RedisCacheOptions> redisCacheOptions,
+                            Optional<RedisClientDelegate> redisClientDelegate,
                             ApplicationContext applicationContext,
                             ProjectClustersService projectClustersService,
                             ProjectClustersCachingAgentProperties projectClustersCachingAgentProperties) {
-    return new CoreProvider([
-      new CleanupPendingOnDemandCachesAgent(redisCacheOptions, redisClientDelegate, applicationContext),
+    List<Agent> agents = [
       new ProjectClustersCachingAgent(
         projectClustersService,
         projectClustersCachingAgentProperties
       )
-    ])
+    ]
+
+    if (redisCacheOptions.isPresent() && redisClientDelegate.isPresent()) {
+      agents.add(new CleanupPendingOnDemandCachesAgent(redisCacheOptions, redisClientDelegate, applicationContext))
```

Comment:
You'll need to `redisCacheOptions.get()` and `redisClientDelegate.get()` here.

Reply:
Groovy, why you let me be an idiot?